### PR TITLE
Replace setup-scala with setup-java

### PR DESCRIPTION
**Pull request by Philippus**

### CODE REVIEW

**Comment by vincenzobaz** [comment]
Hello @Philippus, thank you for the PR.
Why do you think it is better to use `setup-java`?
My main concern is that it might not setup sbt, however the action seem to succeed... How is it possible?

**Comment by Philippus** [comment]
Some background can be found here: https://github.com/olafurpg/setup-scala/issues/49

**Review by SethTisue — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -21,7 +24,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: olafurpg/setup-scala@v13
+      - uses: actions/setup-java@v2
+        with:
+          distribution: temurin
```
Comment (sideeffffect) [inline .github/workflows/publish.yml:29]: Maybe `java-version: 8` as well?

**Comment by sideeffffect** [comment]
Maybe we should direct users towards https://github.com/japgolly/setup-everything-scala from @japgolly instead? It seems more featurefull.

https://old.reddit.com/r/scala/comments/s5h3wb/should_there_be_a_single_setupscala_github_action/?#hszd7jl

File: .github/workflows/publish.yml (hunk 1 above)
Comment (julienrf) [reply .github/workflows/publish.yml:29]: I believe you need to add `java-version` here as well.

**Review by julienrf — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (Philippus) [reply .github/workflows/publish.yml:29]: Done

**Review by julienrf — APPROVED** [review]

**Pull request merged**